core/src/main/kotlin/com/uber/sdk2/core/ui/UberButton.kt

```diff
@@ -0,0 +1,257 @@
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.graphics.Color
+import android.graphics.Typeface
+import android.util.AttributeSet
+import android.view.Gravity
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.annotation.StyleRes
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.widget.AppCompatButton
+import com.uber.sdk2.core.R
+
+/** [android.widget.Button] that can be used as a button and provides default Uber styling. */
+open class UberButton
+@JvmOverloads
+constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0,
+  defStyleRes: Int = R.style.UberButton,
+) : AppCompatButton(context, attrs, defStyleAttr) {
+  @DrawableRes private var backgroundResource = 0
+
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   * @param attrs attributes for the view.
+   * @param defStyleAttr the default attribute to use for a style if none is specified.
+   * @param defStyleRes the default style, used only if defStyleAttr is 0 or can not be found in the
+   *   theme.
+   */
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   * @param attrs attributes for the view.
+   * @param defStyleAttr the default attribute to use for a style if none is specified.
+   */
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   * @param attrs attributes for the view.
+   */
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   */
+  init {
+    val uberStyle =
+      UberStyle.getStyleFromAttribute(
+        context,
+        attrs,
+        defStyleRes,
+        R.styleable.UberButton,
+        R.styleable.UberButton_ub__style,
+      )
+    init(context, 0, attrs, defStyleAttr, uberStyle)
+  }
+
+  protected open fun init(
+    context: Context,
+    @StringRes defaultText: Int,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    uberStyle: UberStyle,
+  ) {
+    val defStyleRes = STYLES[uberStyle.value]
+
+    applyStyle(context, defaultText, attrs, defStyleAttr, defStyleRes)
+  }
+
+  protected fun applyStyle(
+    context: Context,
+    @StringRes defaultText: Int,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    @StyleRes defStyleRes: Int,
+  ) {
+    setBackgroundAttributes(context, attrs, defStyleAttr, defStyleRes)
+    setDrawableAttributes(context, attrs, defStyleAttr, defStyleRes)
+    setPaddingAttributes(context, attrs, defStyleAttr, defStyleRes)
+    setTextAttributes(context, attrs, defStyleAttr, defStyleRes)
+    if (defaultText != 0) {
+      setText(defaultText)
+    }
+  }
+
+  protected val activity: Activity
+    get() {
+      var context = context
+      while (context !is Activity && context is ContextWrapper) {
+        context = context.baseContext
+      }
+
+      if (context is Activity) {
+        return context
+      }
```

Comment:
Move if context is Activity before the while check instead of checking constantly if the context is an instance of Activity. Also I would change is -> instanceof

Reply:
I would have to put an if statement after the while in either case. Not sure how it helps?
`istanceof` can be used in java. In kotlin I need `is`